jolie/src/jolie/runtime/typing/TypePair.java

```diff
@@ -0,0 +1,84 @@
+package jolie.runtime.typing;
+
+import jolie.lang.NativeType;
+import jolie.runtime.Value;
+import jolie.util.Range;
+
+import java.util.Map;
+
+/**
+ * Created by User on 01.08.2015.
+ */
+public class TypePair extends Type {
+    public Type getT1() {
```

Comment:
How about left() and right() instead of getT1 and getT2? This way, we get an intuitive name that maps to the syntax (where we have a left and right type).